<?xml version="1.0"?>
 
<launch>
  <!-- Calibration -->
  <node pkg="lidar_calibration" type="lidar_calibration_node" name="lidar_calibration_node" output="screen" args="">
    <param name="actuator_frame" value="lidar_actuator_frame"/>
    <param name="rotation_offset_pitch" value="-1.57"/>
    <param name="detect_ground_plane" value="false"/> <!-- spinning axis not parallel to ground -->
    <param name="detect_ceiling" value="false"/>
    <param name="ground_frame" value=""/>
    
    <!-- these have to be part of URDF (not just tf) -->
    <param name="o_spin_frame" value="spin_lidar_lidar_mount_link"/> <!-- spinning frame -->
    <param name="o_laser_frame" value="spin_lidar_vertical_hokuyo_laser_frame"/> <!-- laser frame-->
    <param name="tf_wait_duration" value="10"/>
     
    <param name="save_calibration" value="false"/> 
    <param name="save_path" value="$(find argo_tracker_description)/calibrations/$(optenv ARGO_ROBOT_TYPE atex_robot)_spinning_vertical_lidar_calibration.urdf.xacro"/>
    
    <remap from="request_scan1" to="/pitch_lidar/request_scan" />
    <remap from="request_scan2" to="/vertical_lidar/request_scan" />
    <remap from="reset_cloud1" to="/pitch_lidar/reset_cloud" />
    <remap from="reset_cloud2" to="/vertical_lidar/reset_cloud" />
  </node>
</launch>